lib/url-helpers.js

```diff
@@ -166,3 +175,11 @@ export const isTrustedRedirectHost = host => {
     return host === domain || host.endsWith(`.${domain}`);
   });
 };
+
+export const addParentToURLIfMissing = (router, account, url) => {
+  const query = router.query;
+  const urlWithParent = getCollectivePageRoute(account) + url;
+  if ([CollectiveType.EVENT, CollectiveType.PROJECT].includes(account?.type) && !query.parentCollectiveSlug) {
+    router.push(urlWithParent, undefined, { shallow: true });
```

Comment:
:ice_cream: Sounds like we don't need to build the URL if not redirecting
```suggestion
  if ([CollectiveType.EVENT, CollectiveType.PROJECT].includes(account?.type) && !query.parentCollectiveSlug) {
    const urlWithParent = getCollectivePageRoute(account) + url;
    router.push(urlWithParent, undefined, { shallow: true });
```